### PR TITLE
feat(bundler): allow the use of external library CSS

### PR DESCRIPTION
**Pull request by JamesSingleton**
This has been tested locally with `semantic-ui-react` & `semantic-ui-css`, `react-bootstrap`, `bulma`, `material-ui`. Unfortunately I did have to disable `purgecss` for everything to work because it was removing styles that were being used by components.

Any `import styles from './RootStyles.scss';` includes this code
```css
.test {
  background-color: blue;
}
```

### Sample Code
```jsx
import React from 'react';
import {
  Button, Container, Divider, Grid, Header, Image, Menu, Segment,
} from 'semantic-ui-react';
import 'semantic-ui-css/semantic.min.css';
import childRoutes from '../childRoutes';
import styles from './RootStyles.scss';

const SemanticUiRoot = () => (
  <Container style={{ marginTop: '3em' }}>
    <h1 className={styles.test}>TESTING</h1>
    <Header as="h1">Theming Examples</Header>

    <Header as="h2" dividing={true}>
      Site
    </Header>

    <Grid columns={3} stackable={true}>
      <Grid.Column>
        <Header as="h1">Heading 1</Header>
        <Header as="h2">Heading 2</Header>
        <Header as="h3">Heading 3</Header>
        <Header as="h4">Heading 4</Header>
        <Header as="h5">Heading 5</Header>

        <p>
          Nullam quis risus eget urna mollis ornare vel eu leo. Cum sociis natoque penatibus et
          magnis dis parturient montes, nascetur ridiculus mus. Nullam id dolor id nibh ultricies
          vehicula.
        </p>
      </Grid.Column>

      <Grid.Column>
        <Header as="h2">Example body text</Header>

        <p>
          Nullam quis risus eget <a href="#">urna mollis ornare</a> vel eu leo. Cum sociis natoque
          penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nullam id dolor id nibh
          ultricies vehicula.
        </p>
        <p>
          <small>This line of text is meant to be treated as fine print.</small>
        </p>
        <p>
          The following snippet of text is <strong>rendered as bold text</strong>.
        </p>
        <p>
          The following snippet of text is <em>rendered as italicized text</em>.
        </p>
        <p>
          An abbreviation of the word attribute is <abbr title="attribute">attr</abbr>.
        </p>
      </Grid.Column>

      <Grid.Column>
        <Grid
          centered={true}
          columns={3}
          padded={true}
          stackable={true}
          style={{ margin: '-1.5em', width: 400 }}
          textAlign="center"
        >
          <Grid.Column color="red" style={{ margin: '0.5em', height: 50 }}>
            Red
          </Grid.Column>
          <Grid.Column color="orange" style={{ margin: '0.5em', height: 50 }}>
            Orange
          </Grid.Column>
          <Grid.Column color="yellow" style={{ margin: '0.5em', height: 50 }}>
            Yellow
          </Grid.Column>
          <Grid.Column color="olive" style={{ margin: '0.5em', height: 50 }}>
            Olive
          </Grid.Column>
          <Grid.Column color="green" style={{ margin: '0.5em', height: 50 }}>
            Green
          </Grid.Column>
          <Grid.Column color="teal" style={{ margin: '0.5em', height: 50 }}>
            Teal
          </Grid.Column>
          <Grid.Column color="blue" style={{ margin: '0.5em', height: 50 }}>
            Blue
          </Grid.Column>
          <Grid.Column color="violet" style={{ margin: '0.5em', height: 50 }}>
            Violet
          </Grid.Column>
          <Grid.Column color="purple" style={{ margin: '0.5em', height: 50 }}>
            Purple
          </Grid.Column>
          <Grid.Column color="pink" style={{ margin: '0.5em', height: 50 }}>
            Pink
          </Grid.Column>
          <Grid.Column color="brown" style={{ margin: '0.5em', height: 50 }}>
            Brown
          </Grid.Column>
          <Grid.Column color="grey" style={{ margin: '0.5em', height: 50 }}>
            Grey
          </Grid.Column>
          <Grid.Column color="black" style={{ margin: '0.5em', height: 50 }}>
            Black
          </Grid.Column>
        </Grid>
      </Grid.Column>
    </Grid>

    <Header as="h2" dividing={true}>
      Menu
    </Header>

    <Grid columns={3} doubling={true}>
      <Grid.Column>
        <Menu
          items={[
            { key: '1', name: 'link-1', content: 'Link' },
            { key: '2', name: 'link-2', content: 'Link' },
            { key: '3', name: 'link-3', content: 'Link' },
          ]}
          pointing={true}
          secondary={true}
        />
      </Grid.Column>

      <Grid.Column>
        <Menu
          items={[
            { key: '1', name: 'link-1', content: 'Link' },
            { key: '2', name: 'link-2', content: 'Link' },
            { key: '3', name: 'link-3', content: 'Link' },
          ]}
          pointing={true}
          tabular={true}
        />
      </Grid.Column>

      <Grid.Column>
        <Menu
          items={[
            { key: 'l1', name: 'link-1', content: 'Link' },
            { key: 'l2', name: 'link-2', content: 'Link' },
            {
              key: 't1', name: 'text-1', content: 'Right text', position: 'right',
            },
          ]}
          pointing={true}
        />
      </Grid.Column>
    </Grid>

    <Header as="h2" dividing={true}>
      Buttons
    </Header>

    <Grid columns="equal">
      <Grid.Column>
        <Button>Default</Button>
        <Button primary={true}>Primary</Button>
        <Button secondary={true}>Secondary</Button>
        <Button basic={true}>Basic</Button>
        <Button compact={true}>Compact</Button>

        <Divider />

        <Button icon="heart" />
        <Button content="Labeled" icon="heart" labelPosition="left" />
        <Button content="Labeled" icon="heart" labelPosition="right" />

        <Divider />

        <Button.Group>
          <Button>Combo</Button>
        </Button.Group>

        <Divider />

        <Button animated={true}>
          <Button.Content visible={true}>Horizontal</Button.Content>
          <Button.Content hidden={true}>Hidden</Button.Content>
        </Button>
        <Button animated="vertical">
          <Button.Content visible={true}>Vertical</Button.Content>
          <Button.Content hidden={true}>Hidden</Button.Content>
        </Button>
        <Button animated="fade">
          <Button.Content visible={true}>Fade In</Button.Content>
          <Button.Content hidden={true}>Hidden</Button.Content>
        </Button>

        <Divider />

        <Button disabled={true}>Disabled</Button>
        <Button loading={true}>Loading</Button>

        <Divider />

        <Button.Group>
          <Button>1</Button>
          <Button>2</Button>
          <Button>3</Button>
        </Button.Group>

        <Button.Group>
          <Button icon="align left" />
          <Button icon="align center" />
          <Button icon="align right" />
          <Button icon="align justify" />
        </Button.Group>

        <Button.Group>
          <Button>1</Button>
          <Button.Or />
          <Button>2</Button>
        </Button.Group>

        <Divider />

        <Button.Group attached="top" widths={2}>
          <Button>One</Button>
          <Button>Two</Button>
        </Button.Group>
        <Button.Group attached="bottom" widths={2}>
          <Button>One</Button>
          <Button>Two</Button>
        </Button.Group>
      </Grid.Column>

      <Grid.Column>
        <Button size="mini">Mini</Button>
        <Button size="tiny">Tiny</Button>
        <Button size="small">Small</Button>
        <Button size="large">Large</Button>
        <Button size="big">Big</Button>
        <Button size="huge">Huge</Button>
        <Button size="massive">Massive</Button>

        <Divider />

        <Button color="yellow" style={{ marginBottom: '1em' }}>
          Yellow
        </Button>
        <Button color="orange" style={{ marginBottom: '1em' }}>
          Orange
        </Button>
        <Button color="green" style={{ marginBottom: '1em' }}>
          Green
        </Button>
        <Button color="teal" style={{ marginBottom: '1em' }}>
          Teal
        </Button>
        <Button color="blue" style={{ marginBottom: '1em' }}>
          Blue
        </Button>
        <Button color="purple" style={{ marginBottom: '1em' }}>
          Purple
        </Button>
        <Button color="pink" style={{ marginBottom: '1em' }}>
          Pink
        </Button>
        <Button color="red" style={{ marginBottom: '1em' }}>
          Red
        </Button>
        <Button color="black" style={{ marginBottom: '1em' }}>
          Black
        </Button>

        <Divider />

        <Segment inverted={true}>
          <Button inverted={true}>Inverted</Button>
          <Button basic={true} inverted={true}>
            Basic
          </Button>
          <Button color="blue" inverted={true}>
            Colored
          </Button>
          <Button basic={true} color="blue" inverted={true}>
            Basic Colored
          </Button>
        </Segment>
      </Grid.Column>
    </Grid>
  </Container>
);

// Read about childRoutes:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/Routing.md#childroutes
SemanticUiRoot.childRoutes = childRoutes;

// Read about appConfig:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/App-Configuration.md
/* istanbul ignore next */
if (!global.BROWSER) {
  // eslint-disable-next-line global-require
  SemanticUiRoot.appConfig = require('../appConfig').default;
}

export default SemanticUiRoot;
```

### Screenshot
semantic-ui

### Sample Code

```jsx
import React, { useState, Fragment } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Toast from 'react-bootstrap/Toast';
import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import childRoutes from '../childRoutes';
import styles from './RootStyles.scss';

const ExampleToast = ({ children }) => {
  const [show, toggleShow] = useState(true);

  return (
    <Fragment>
      {!show && <Button onClick={() => toggleShow(true)}>Show Toast</Button>}
      <Toast show={show} onClose={() => toggleShow(false)}>
        <Toast.Header>
          <strong className="mr-auto">React-Bootstrap</strong>
        </Toast.Header>
        <Toast.Body>{children}</Toast.Body>
      </Toast>
    </Fragment>
  );
};

const SemanticUiRoot = () => (
  <Container className="p-3">
    <Jumbotron>
      <h1 className={styles.header}>Welcome To React-Bootstrap</h1>
      <ExampleToast>
        We now have Toasts
        <span role="img" aria-label="tada">
          🎉
        </span>
      </ExampleToast>
    </Jumbotron>
  </Container>
);

// Read about childRoutes:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/Routing.md#childroutes
SemanticUiRoot.childRoutes = childRoutes;

// Read about appConfig:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/App-Configuration.md
/* istanbul ignore next */
if (!global.BROWSER) {
  // eslint-disable-next-line global-require
  SemanticUiRoot.appConfig = require('../appConfig').default;
}

export default SemanticUiRoot;
```

### Screenshot
react-bootstrap

### Sample Code
```jsx
import React from 'react';
import 'bulma/css/bulma.min.css';
import childRoutes from '../childRoutes';

const SemanticUiRoot = () => (
  <div className="container">
    <div className="notification">
      This container is <strong>centered</strong> on desktop.
    </div>
    <div className="columns">
      <div className="column is-one-third">
        <div className="card">
          <div className="card-image">
            <figure className="image is-4by3">
              <img src="https://bulma.io/images/placeholders/1280x960.png" alt="Placeholder image" />
            </figure>
          </div>
          <div className="card-content">
            <div className="media">
              <div className="media-left">
                <figure className="image is-48x48">
                  <img src="https://bulma.io/images/placeholders/96x96.png" alt="Placeholder image" />
                </figure>
              </div>
              <div className="media-content">
                <p className="title is-4">John Smith</p>
                <p className="subtitle is-6">@johnsmith</p>
              </div>
            </div>

            <div className="content">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              Phasellus nec iaculis mauris. <a>@bulmaio</a>.
              <a href="#">#css</a> <a href="#">#responsive</a>
              <br />
              <time dateTime="2016-1-1">11:09 PM - 1 Jan 2016</time>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);

// Read about childRoutes:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/Routing.md#childroutes
SemanticUiRoot.childRoutes = childRoutes;

// Read about appConfig:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/App-Configuration.md
/* istanbul ignore next */
if (!global.BROWSER) {
  // eslint-disable-next-line global-require
  SemanticUiRoot.appConfig = require('../appConfig').default;
}

export default SemanticUiRoot;
```

### Screenshot
bulma

### Sample Code

```jsx
import React, {Fragment } from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import childRoutes from '../childRoutes';
import styles from './RootStyles.scss';

const SemanticUiRoot = () => (
  <Fragment>
    <CssBaseline />
    <Container maxWidth="sm">
      <Typography component="div" style={{ backgroundColor: '#cfe8fc', height: '100vh' }} />
    </Container>
  </Fragment>
);

// Read about childRoutes:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/Routing.md#childroutes
SemanticUiRoot.childRoutes = childRoutes;

// Read about appConfig:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/App-Configuration.md
/* istanbul ignore next */
if (!global.BROWSER) {
  // eslint-disable-next-line global-require
  SemanticUiRoot.appConfig = require('../appConfig').default;
}

export default SemanticUiRoot;
```

### Screenshot
material-ui

### CODE REVIEW

File: packages/one-app-bundler/webpack/loaders/common.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ const cssLoader = ({ name = '', importLoaders = 2 } = {}) => ({
     importLoaders,
     modules: {
       localIdentName: `${name && `${name}__`}[name]__[local]___[hash:base64:5]`,
+      // eslint-disable-next-line max-params, no-unused-vars
+      getLocalIdent: (loaderContext, localIdentName, localName, options) => (
```
Comment (mtomcal) [inline packages/one-app-bundler/webpack/loaders/common.js:29]: Im not sure what this is for as it seems super bundler specific. Can we get some code comments to describe the purpose of this function added?

File: packages/one-app-bundler/webpack/loaders/ssr-css-loader/index.js [hunk 2]
```diff
@@ -20,8 +20,12 @@ const cssBasePathString = JSON.stringify(path.resolve(__dirname, 'css-base.js'))
 
 const CSS_LOADER_FINDER = /var ___CSS_LOADER_API_IMPORT___ = (__webpack_){0,1}require(__){0,1}\([.a-zA-Z0-9/_*!\s-]*"[.a-zA-Z0-9/_]+\/css-loader\/dist\/runtime\/api.js"\);\n\s*exports = ___CSS_LOADER_API_IMPORT___\((undefined|false)?\);/;
 
+const CSS_RUNTIME_FINDER = /var ___CSS_LOADER_API_IMPORT___ = (__webpack_){0,1}require(__){0,1}\([.a-zA-Z0-9/_*!\s-]*"[.a-zA-Z0-9/_]+\/css-loader\/dist\/runtime\/api.js"\);/;
+
+const CSS_EXPORTS_FINDER = /exports = ___CSS_LOADER_API_IMPORT___\((undefined|false)?\);/;
```
Comment (mtomcal) [inline packages/one-app-bundler/webpack/loaders/ssr-css-loader/index.js:25]: Would you add code comments here on specifics of the purpose of these regexes? Regexes can be notoriously hard to reason about without context in advanced cases.

File: packages/one-app-bundler/__tests__/webpack/loaders/common.spec.js [hunk 3]
```diff
@@ -57,6 +57,38 @@ describe('Common webpack loaders', () => {
       expect(config.options.modules.localIdentName).toBe('[name]__[local]___[hash:base64:5]');
       expect(config).toMatchSnapshot();
     });
+    it('should still return locaName from getLocalIdent if resourcePath includes node_modules', () => {
```
Comment (mtomcal) [inline packages/one-app-bundler/__tests__/webpack/loaders/common.spec.js:60]: `locaName` to `localName`?

**Comment by mtomcal** [comment]
Not in scope of this effort but can we add your example modules with foreign CSS inside a folder and run some snapshot testing of some to verify it bundled the CSS correctly? Add integration style tests to bundler?

File: packages/one-app-bundler/__tests__/webpack/loaders/common.spec.js (hunk 3 above)
Comment (JamesSingleton) [reply packages/one-app-bundler/__tests__/webpack/loaders/common.spec.js:60]: Good catch!

**Pull request merged**